### PR TITLE
Lean 2651

### PR DESCRIPTION
**Pull request by asouqi**

### CODE REVIEW

**Comment by mnatsheh** [comment]
@asouqi do you think it makes sense to have the locales here directly rather than exposed in the api?

**Comment by asouqi** [comment]
> @asouqi do you think it makes sense to have the locales here directly rather than exposed in the api?

that will be ok in case we are going to support just one or a couple of languages, but if we need to [support all](https://github.com/jgm/citeproc/tree/master/locales) of them it's to add to the app size 1.2 MB

**Comment by mnatsheh** [comment]
I suggest we support a single locale by default here. If we need any more, we need to pass the locale XML through parameters the same way we send in the style. What do you think?

**Comment by asouqi** [comment]
> I suggest we support a single locale by default here. If we need any more, we need to pass the locale XML through parameters the same way we send in the style. What do you think?

that is a great idea, update it based on that.

File: src/CitationProvider.ts [hunk 1]
```diff
@@ -14,36 +14,40 @@
...
 
 import { variableWrapper } from './citeproc-variable-wrapper'
 import { convertBibliographyItemToCSL } from './csl-converter'
+import defaultLocale from './defaultLocale'
 
 interface Props {
   getLibraryItem: (id: string) => BibliographyItem | undefined
   citationStyle: string
-  lang?: string // Default en-GB
+  lang?: string // Default en-US
```
Comment (mnatsheh) [inline src/CitationProvider.ts]: Is this needed anymore?

File: src/CitationProvider.ts [hunk 2]
```diff
@@ -14,36 +14,40 @@
...
+  locale: string | null
```
Comment (mnatsheh) [inline src/CitationProvider.ts]: Would `locale?: string` work here?

Comment (asouqi) [reply src/CitationProvider.ts]: updated

File: src/CitationProvider.ts (hunk 1 above)
Comment (asouqi) [reply src/CitationProvider.ts]: will use it in [citeproc](https://github.com/Atypon-OpenSource/manuscripts-library/blob/master/src/CitationProvider.ts#L50)

Comment (mnatsheh) [reply src/CitationProvider.ts]: Yes, but does the value matter in our case? citeproc will just use `retrieveLocale` to get the locale config, so can we just pass a static value ("en" for example) to citeproc?

Comment (asouqi) [reply src/CitationProvider.ts]: updated

**Review by mnatsheh — APPROVED** [review]

**Pull request merged**